riscv/processor.h

```diff
@@ -83,6 +83,7 @@ struct state_t
   // control and status registers
   std::unordered_map<reg_t, csr_t_p> csrmap;
   reg_t prv;    // TODO: Can this be an enum instead?
+  reg_t prev_prv;
```

Comment:
We can use the existing variable "last_inst_priv" of the same struct "state_t".

Reply:
Updated per the suggestion. Note that since `last_inst_priv` was being updated in the logging path, the code was changed to make the update unconditional in both the fast and logged execution paths.

Reply:
> Updated per the suggestion. Note that since `last_inst_priv` was being updated in the logging path, the code was changed to make the update unconditional in both the fast and logged execution paths.

This sounds like a bug fix if the `last_inst_priv` wasn't being updated in the fast path before?

If I understand correctly, the bug fix should be a separate commit explaining that change in isolation.

Reply:
That is correct. Please see if the update (and new commit) reflects the intent of the change.